src/Symfony/Component/Uid/BinaryUtil.php

```diff
@@ -142,4 +142,34 @@ public static function timeToDateTime(string $time): \DateTimeImmutable
 
         return \DateTimeImmutable::createFromFormat('U.u?', substr_replace($time, '.', -7, 0));
     }
+
+    /**
+     * @return string Count of 100-nanosecond intervals since the UUID epoch 1582-10-15 00:00:00 in hexadecimal
+     */
+    public static function dateTimeToHex(\DateTimeInterface $time): string
+    {
+        if (\PHP_INT_SIZE >= 8) {
+            if (-self::TIME_OFFSET_INT > $time = (int) $time->format('Uu0')) {
+                throw new \InvalidArgumentException('The provided UUID timestamp must be higher than 1582-10-15.');
```

Comment:
This error message is confusing because:

* The user provides a DateTimeInterface time, not a timestamp
* The error says that timestamp must be higher than `1582-10-15`, but that's not possible because we're comparing timestamps with formatted dates.

We could change it as follows:

```php
throw new \InvalidArgumentException('The given UUID date cannot be earlier than 1582-10-15.');
```

Same for other similar error messages in this PR.

Reply:
updated, thanks for the suggestion